keg_elements/db/mixins.py

```diff
@@ -67,30 +67,116 @@ def from_dict(self, data):
+
+        .. note: By default hybrid properties are not included in the returned dict. To add a hybrid
+        property to the returned dict pass a list of the property names and they will be included.
+        """
+        data = dict([(name, getattr(self, name))
+                     for name in self.column_names()
+                     if name not in exclude])
+
+        for hybrid in hybrids:
+            data[hybrid] = getattr(self, hybrid)
+
+        return data
+
+    @might_commit
+    @might_flush
+    @classmethod
+    def add(cls, **kwargs):
+        obj = cls()
+        obj.from_dict(kwargs)
+        db.session.add(obj)
+        return obj
+
+    @might_commit
+    @might_flush
+    @classmethod
+    def delete(cls, oid):
+        """Delete an object from the session
+
+        :param oid: the object identifier, normally the primary key
+        :rtype: bool
+        :return: The result of the operation
+        """
+        obj = cls.query.get(oid)
+
+        if obj is None:
+            return False
+
+        db.session.delete(obj)
+        return True
+
+    @might_commit
+    @classmethod
+    def delete_cascaded(cls):
+        cls.query.delete(synchronize_session=False)
+        db.session.expire_all()
+
+    @classmethod
+    def column_names(cls):
+        return {col.key for col in cls.__mapper__.columns}
 
+    @might_commit
+    @might_flush
     @classmethod
-    def add(cls, _commit=True, _flush=False, **kwargs):
-        o = cls()
-        o.from_dict(kwargs)
-        db.session.add(o)
-        if _flush:
-            session_flush()
-        elif _commit:
-            session_commit()
-        return o
+    def edit(cls, oid=None, **kwargs):
+        try:
+            oid = oid or kwargs.pop('id')
+        except KeyError:
+            raise AttributeError('No primary key was found in `oid` or `kwargs`'
+                                 ' for which to retrieve the object to edit')
+
+        obj = cls.query.get(oid)
+        obj.from_dict(kwargs)
+        return obj
 
     @classmethod
-    def delete_all(cls, commit=True):
-        retval = cls.query.delete()
-        if commit:
-            session_commit()
-        return retval
+    def get_by(cls, **kwargs):
+        """Returns the instance of this class matching the given criteria or
+        None if there is no record matching the criteria.
+
+        If multiple records are returned, an exception is raised.
+        """
+        return cls.query.filter_by(**kwargs).one_or_none()
+
+    @classmethod
+    def get_where(cls, *clauses):
+        """
+        Returns the instance of this class matching the given clause(s) or None
+        if there is no record matching the criteria.
+
+        If multiple records are returned, an exception is raised.
+        """
+        return cls.query.filter(*clauses).one_or_none()
+
+    @classmethod
+    def pairs(cls, key_field, value_field, order_by=list(), query=None,
```

Comment:
I believe `list()` is still mutable, so you'll want to use `None` or `()` I think.